dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/tests/Audacia.CodeAnalysis.Analyzers.Test/Rules/NestedControlStatementsAnalyzerTests.cs

```diff
@@ -192,6 +190,43 @@ static void Main(string[] args)
             VerifyDiagnostic(test, expected);
         }
 
+        [TestMethod]
+        public void Diagnostics_For_Method_Body_With_More_Than_Max_Allowed_If_Statements_With_Else_Clause()
```

Comment:
I think another test to add that would more closely represent the bug would be the following structure:

```csharp
if (true)
{
    if (true)
    {
    }
    else
    {
        if (true)
        {
        }
    }
}
```

In this case the final `if` (line 8) would be flagged even though it is inside the `else`

Reply:
I've added a case for this. It's required some logic changes.